File: Java/BestTimeToBuyAndSellStock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProf = 0, j = 0;
        for (int i = 1; i < prices.length; i ++){
          int compare = prices[i] - prices[j];
          if(compare > maxProf){
            maxProf = compare;
          }     
          if(prices[i] < prices[j])
              j = i;
        }
        return maxProf;
    }
}

File: Rust/best-time-to-buy-and-sell-stock.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut min = i32::MAX;
    let mut max = 0;
    for i in prices{
        if i < min {
            min = i;
        } else if i - min > max {
            max = i - min;
        }
    }
    max
    }
}
